From e6738c3fb4735300a6f394e0c0d89c7763583fa5 Mon Sep 17 00:00:00 2001
From: Jan Kiszka <jan.kiszka@siemens.com>
Date: Wed, 3 Nov 2021 14:09:15 +0100
Subject: [PATCH 9/9] boards: siemens: iot2050: Ignore network errors during
 bootstage tracking

We need to filter out NET_ETH_START errors because we have to enable
networking in order to propagate the MAC addresses to the DT while there
is no network driver for the prueth in U-Boot yet.

Signed-off-by: Jan Kiszka <jan.kiszka@siemens.com>
---
 board/siemens/iot2050/board.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/board/siemens/iot2050/board.c b/board/siemens/iot2050/board.c
index b2110978ae..817fbf4ecb 100644
--- a/board/siemens/iot2050/board.c
+++ b/board/siemens/iot2050/board.c
@@ -259,7 +259,8 @@ void show_boot_progress(int progress)
 	struct udevice *dev;
 	int ret;
 
-	if (progress < 0 || progress == BOOTSTAGE_ID_ENTER_CLI_LOOP) {
+	if ((progress < 0 && progress != -BOOTSTAGE_ID_NET_ETH_START) ||
+	    progress == BOOTSTAGE_ID_ENTER_CLI_LOOP) {
 		ret = led_get_by_label("status-led-green", &dev);
 		if (ret == 0)
 			led_set_state(dev, LEDST_OFF);
-- 
2.31.1

